### ISSUE TITLE
Problem of sortText in completion item

### ISSUE THREAD

**Issue by chemzqm** [castwide/solargraph#85]
When completiong method names by `.` I have:
```
  {
    "word": "bar",
    "menu": "[LS]",
    "kind": "Method",
    "sortText": "bar0000",
    "filterText": "bar",
    "isSnippet": false,
    "abbr": "bar",
    "user_data": "{\"cid\":1536218759,\"source\":\"solargraph\"}",
    "score": 0.092
  }
```
The `\0` byte should prepend to the text instead of append to make it have higher priority for shown, so it should be `\0bar`

**Comment by castwide** [comment]
Do you have a test case where this causes a problem?

The numbering at the end of `sortText` is intended to ensure that the list keeps duplicate names in their order of precedence. Given the following code:

```Ruby
class Superclass
  def meth; end
end
class Subclass < Superclass
  def meth; end
end
```

Completions for `Subclass.new.` should show `Subclass#meth` before `Superclass#meth`. Their `sortText` values would be `meth0000` and `meth0001` respectively. Without the `sortText` values, clients might change the order while filtering the results.

**Comment by chemzqm** [comment]
<img width="310" alt="screen shot 2018-09-06 at 9 03 48 pm" src="https://user-images.githubusercontent.com/251450/45159074-651a9400-b218-11e8-9c84-774b411db30c.png">

This is what I get with latest solargraph extension in VSCode, could be better if `bar` comes first which have `\0` appended in `sortText`

**Comment by castwide** [comment]
Gotcha. I'm glad you brought that up. I've been experimenting with sorting rules that do something like what you're suggesting, and I'm inclined to agree that it's better.

In the current sorting rules, all methods are sorted alphabetically, and matching names are sorted in order of precedence.

In the new sorting rules, all methods are sorted by namespace precedence first, then alphabetically within the namespace. That would put them in the order you're proposing, with a couple of additional caveats. Example:

![image](https://user-images.githubusercontent.com/3704922/45161147-2dc8ca80-b1b9-11e8-9a04-fb40523ba1e5.png)

The namespace order becomes `Sub`, `Sup`, `Object`, `BasicObject`.

I pushed this change to master and expect to include it in gem 0.27.0. Suggestions and feedback are welcome.

**Comment by castwide** [comment]
The new sorting rules are published in gem 0.27.0.

**Comment by chemzqm** [comment]
It works, thanks.
